collect_app/src/main/java/org/odk/collect/android/preferences/screens/AdminPreferencesFragment.java

```diff
@@ -153,12 +157,20 @@ public boolean onPreferenceClick(Preference preference) {
                     startActivity(pref);
                     break;
                 case DELETE_PROJECT_KEY:
-                    new AlertDialog.Builder(requireActivity())
-                            .setTitle(R.string.delete_project_confirm_message)
-                            .setNegativeButton(R.string.delete_project_no, (dialog, which) -> {
-                            })
-                            .setPositiveButton(R.string.delete_project_yes, (dialog, which) -> deleteProject())
-                            .show();
+                    if (instancesRepositoryProvider.get().getAll().isEmpty()) {
```

Comment:
Actually, I just realized: as we still want to be able to delete projects with sent instances, this isn't quite right. 

Maybe it's worth having a helper method for if you can delete projects or alternatively return a specific error from `ProjectDeleter#deleteCurrentProject` so you don't have to test both deleted and sent projects at the Espresso level?

Reply:
> Actually, I just realized: as we still want to be able to delete projects with sent instances, this isn't quite right.

Ahh right good catch!

Reply:
Ok I came up with a new implementation could you take a look?

I have one problem... Have you tried using espresso with preference fragments? I can't seem to get it working. I wanted to perform simple tests that I click that `Delete project` option and check displayed dialogs using
`onView(withText(R.string.delete_project)).perform(scrollTo(), click())`
doesn't work. I tried different themes to no avail.

Reply:
Oh I think the preferences fragments might use `RecyclerView` under the hood so you probably need to use the specific actions it needs. We have `Page#scrollToRecyclerViewItemAndClickText` for that. Maybe it's worth making an abstract `Page` subclass for "settings" pages with helpers like `clickSetting` or something, so we only have to update the helpers in one place when they inevitably introduce a new list view...

Reply:
First of all it's in `collect_app/androidTest` so we need to move it to `testshared` probably and put in a package that will be shared between both types of tests.
When it comes to that method I tried it but to no avail. I think we need to create separate issues for that, one for moving those espresso helpers and another one for investigating how to use preference fragments in fragment scenario.
What do you think?